router/pump.go

```diff
@@ -347,6 +347,8 @@ func newContainerPump(container *docker.Container, stdout, stderr io.Reader) *co
 	pump := func(source string, input io.Reader) {
 		buf := bufio.NewReader(input)
 		for {
+			// Discard header bytes from log message
+			_, err := buf.Discard(8)
```

Comment:
does it make sense to check the API version and only do this above the appropriate version?

Reply:
I've dug in a little deeper.  Looks like container/{id}/logs is following the same output pattern as the container/{id}/attach endpoint, with headers only being prepended when Tty=false. 

I've confirmed this behavior in docker 17.10.0-ce (API 1.33) and the oldest version I still have running 1.10.3 (API 1.22). From some of the GitHub issues it might have been set to do this since API 1.13 or before.

I'm going to update this and wrap the Discard in a check for > 1.13 API version and look for TTY=False.

References: 
https://docs.docker.com/engine/api/v1.33/#operation/ContainerAttach
https://github.com/moby/moby/issues/7375
https://github.com/moby/moby/issues/14201